packages/viz/src/Treemap/Treemap.tsx

```diff
@@ -0,0 +1,98 @@
+import { forwardRef } from "react";
+import ReactECharts from "echarts-for-react/lib/core";
+import { TreemapChart } from "echarts/charts";
+import {
+  DatasetComponent,
+  GridComponent,
+  LegendComponent,
+  TooltipComponent,
+} from "echarts/components";
+import * as echarts from "echarts/core";
+import { useTheme } from "@hitachivantara/uikit-react-core";
+
+import { HvBaseChart } from "../BaseChart";
+import { HvChartTooltipClasses, useOption, useTooltip } from "../hooks";
+import { HvChartCommonProps } from "../types/common";
+import { HvTreemapData } from "../types/generic";
+
+// Register chart components
+echarts.use([
+  TreemapChart,
+  DatasetComponent,
+  GridComponent,
+  TooltipComponent,
+  LegendComponent,
+]);
+
+export interface HvTreemapChartClasses extends HvChartTooltipClasses {}
+
+export interface HvTreemapChartProps
+  extends Omit<HvChartCommonProps, "data" | "groupBy" | "sortBy" | "grid"> {
+  /** The name of the treemap */
+  name?: string;
+  /** The data to use on the treemap */
+  data?: HvTreemapData[];
+  /** A Jss Object used to override or extend the styles applied to the component. */
+  classes?: HvTreemapChartClasses;
+}
+
+/**
+ * A tree map chart visually represents hierarchical data using nested rectangles, with each rectangle's size proportional to the value it represents.
+ */
+export const HvTreemapChart = forwardRef<ReactECharts, HvTreemapChartProps>(
+  (props, ref) => {
+    const {
+      name,
+      data,
+      classes,
+      width,
+      height,
+      tooltip,
+      onOptionChange,
+      ...others
+    } = props;
+
+    const { colors } = useTheme();
+    const chartTooltip = useTooltip({
+      ...tooltip,
+      type: "single",
```

Comment:
The `type` defined by the user (inside `tooltip`) will be overridden. If there's no `multiple` tooltip type for the treetop, we should remove it.